docs/sources/clients/promtail/configuration.md

```diff
@@ -983,6 +986,47 @@ The list of labels below are discovered when consuming kafka:
 
 To keep discovered labels to your logs use the [relabel_configs](#relabel_configs) section.
 
+### GELF
+
+The `gelf` block configures a GELF UDP listener allowing users to push
+logs to Promtail with the [GELF](https://docs.graylog.org/docs/gelf) protocol.
+Currently only UDP is supported, please submit a feature request if you're interested into TCP support.
+
+> GELF messages can be sent uncompressed or compressed with either GZIP or ZLIB.
+
+Each GELF message received will be encoded in JSON as the log line. For example:
+
+```json
+{"version":"1.1","host":"example.org","short_message":"A short message","timestamp":1231231123,"level":5,"_some_extra":"extra"}
+```
+
+You can leverage [pipeline stages](pipeline_stages) with the GELF target,
+if for example, you want to parse the log line and extract more labels or change the log line format.
+
+```yaml
+# UDP address to listen on. Has the format of "host:port". Default to 0.0.0.0:12201
+listen_address: <string>
+
+# Label map to add to every log message.
+labels:
+  [ <labelname>: <labelvalue> ... ]
+
+# Whether Promtail should pass on the timestamp from the incoming gelf message.
+# When false, or if no timestamp is present on the gelf message, Promtail will assign the current timestamp to the log when it was processed.
+# Default is false
+use_incoming_timestamp: <bool>
+
+```
+
+**Available Labels:**
```

Comment:
I'm suprised you don't expose the extra fields as e.g. `__gelf_message_extra_<name>`

Reply:
Created an issue https://github.com/grafana/loki/issues/4846